src/clj/game/cards/upgrades.clj

```diff
@@ -609,27 +609,28 @@
                    :effect (req (swap! state assoc-in [:runner :register :force-trash] false))}}
 
    "Mwanza City Grid"
-   (let [gain-creds {:req (req (and installed
-                                    this-server
-                                    (pos? (:cards-accessed run 0))))
-                     :silent (req true)
-                     :effect (req (let [cnt (:cards-accessed run)
-                                        total (* 2 cnt)]
-                                    (gain-credits state :corp total)
-                                    (system-msg state :corp
-                                                (str "gains " total " [Credits] from Mwanza City Grid"))))}]
-     {:install-req (req (filter #{"HQ" "R&D"} targets))
-      :events {:pre-access {:req (req (and installed
-                                           ;; Pre-access server is same server as that Mwanza is in the root of
-                                           (= target (second  (:zone card)))))
+   (let [gain-creds-and-clear {:req (req (= (:from-server target) (second (:zone card))))
+                               :silent (req true)
+                               :effect (req (let [cnt (:cards-accessed run)]
+                                              (access-bonus state :runner -3)
+                                              (if cnt ; TODO: this fails if the run featured multiple access phases (e.g. shiro sub firing)
```

Comment:
This needs to be a `when`. Right now, if `cnt` *is* defined, it gains the credits, and if it *isn't* defined, it calls `system-msg`. Change it to a when to only execute both line if `cnt` is defined.